contracts/zaps/OlympusZapV2.sol

```diff
@@ -0,0 +1,198 @@
+
+        // approve "swapTarget" to spend this contracts "fromToken" if needed
+        _approveToken(fromToken, swapTarget, toInvest);
+
+        // swap fromToken -> OHM
+        uint256 tokensBought = _fillQuote(fromToken, OHM, toInvest, swapTarget, swapData);
+
+        // stake OHM for sOHM or gOHM
+        OHMRec = _stake(tokensBought, toToken);
+
+        // Slippage check
+        require(OHMRec > minToToken, "High Slippage");
+
+        emit zapStake(msg.sender, toToken, OHMRec, referral);
+    }
+
+    /// @notice This function acquires Olympus bonds with ETH or ERC20 tokens
+    /// @param fromToken The token used for entry (address(0) if ether)
+    /// @param amountIn The quantity of fromToken being sent
+    /// @param principal The token fromToken is being converted to (i.e. token or LP to bond)
+    /// @param swapTarget Excecution target for the swap or Zap
+    /// @param swapData DEX or Zap data
+    /// @param referral The front end operator address
+    /// @param maxPrice The maximum price at which to buy the bond
+    /// @param bondId The ID of the market
+    /// @return OHMRec The quantity of gOHM due
+    function ZapBond(
+        address fromToken,
+        uint256 amountIn,
+        address principal,
+        address swapTarget,
+        bytes calldata swapData,
+        address referral,
+        uint256 maxPrice,
+        uint256 bondId
+    ) external payable pausable returns (uint256 OHMRec) {
+        // pull users fromToken
+        uint256 toInvest = _pullTokens(fromToken, amountIn, referral, true);
+
+        // make sure "swapTarget" is approved to spend this contracts "fromToken"
+        _approveToken(fromToken, swapTarget, toInvest);
+        // swap fromToken -> bond principal
+        uint256 tokensBought = _fillQuote(
+            fromToken,
+            principal, // to token
+            toInvest,
+            swapTarget,
+            swapData
+        );
+
+        // make sure bond depo is approved to spend this contracts "principal"
+        _approveToken(principal, depo, tokensBought);
+
+        // purchase bond
+        (OHMRec, , ) = IBondDepoV2(depo).deposit(
+            bondId,
+            tokensBought,
+            maxPrice,
+            msg.sender, // depositor
+            referral
+        );
+
+        emit zapBond(msg.sender, principal, OHMRec, referral);
+    }
+
+    ////////////////////////// INTERNAL //////////////////////////
+
+    /// @param amount The quantity of OHM being staked
+    /// @param toToken Either sOHM or gOHM
+    /// @return OHMRec quantity of sOHM or gOHM  received (depending on toToken)
+    function _stake(uint256 amount, address toToken) internal returns (uint256) {
+        uint256 claimedTokens;
+        // approve staking for OHM if needed
+        _approveToken(OHM, staking, amount);
+
+        if (toToken == gOHM) {
+            // stake OHM -> gOHM
+            IStaking(staking).stake(address(this), amount, false, false);
```

Comment:
The last param of this method is `_claim` which removes the need for the additional call in the next line. You can do this for the sOHM call as well.